rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ReportsJdbcRepository.scala

```diff
@@ -382,12 +380,54 @@ class ReportsJdbcRepository(jdbcTemplate : JdbcTemplate) extends ReportsReposito
       }
     }
 
+
+    /*
+     * here, we may have several runs with same nodeId/timestamp
+     * In that case, we need to keep the one with a configId, if such
+     * exists.
+     */
+    def distinctRuns(seq: Seq[AgentRun]): Seq[AgentRun] = {
+      //that one is for a list of agentRun with same id
+      def recDisctinct(runs: List[AgentRun]): AgentRun = {
```

Comment:
How so ? It is an exceptionnal case to have more than one run for one node at a given second. I can't see how it could be more than a few ?